WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/Multi-package/ShippingLabelSinglePackageViewModel.swift

```diff
@@ -246,9 +253,20 @@ final class ShippingLabelSinglePackageViewModel: ObservableObject, Identifiable
             .sink { [weak self] containsHazmat in
                 if !containsHazmat {
                     self?.selectedHazmatCategory = .none
+                } else {
+                    self?.analytics.track(event: .ShippingLabelHazmatDeclaration.containsHazmatChecked())
                 }
             }.store(in: &subscriptions)
 
+        $selectedHazmatCategory
+            .removeDuplicates()
+            .filter { $0 != .none }
+            .sink { [weak self] selectedCategory in
+                guard let order = self?.order else { return }
+                self?.analytics.track(event: .ShippingLabelHazmatDeclaration.hazmatCategorySelected(orderID: order.orderID,
```

Comment:
nit: might look nicer if we guard self here as well to avoid more optional self later
```suggestion
                guard let self else { return }
                self.analytics.track(event: .ShippingLabelHazmatDeclaration.hazmatCategorySelected(orderID: self.order.orderID,
```